spanner/google/cloud/spanner/streamed.py

```diff
@@ -125,12 +155,29 @@ def _merge_values(self, values):
                 self._rows.append(self._current_row)
                 self._current_row = []
 
+    def _restart_iterator(self, _exc_ignored):
+        """Helper for :meth:`consume_next`."""
+        if self._resume_token in (None, b''):
+            raise
+
+        self._response_iterator = self._restart(
+            resume_token=self._resume_token)
+
+    def _bump_iterator(self):
+        """Helper for :meth:`consume_next`."""
+        return six.next(self._response_iterator)
+
     def consume_next(self):
         """Consume the next partial result set from the stream.
 
         Parse the result set into new/existing rows in :attr:`_rows`
+
+        :raises ValueError:
+            if the sleep generator somehow does not yield values.
         """
-        response = six.next(self._response_iterator)
+        response = retry_unavailable(
```

Comment:
I was on the fence about adding more stuff to `retry_target`, but this is just beautiful. Thanks for doing this.